### PR TITLE
Fix sysinfo_test0 for Windows

### PR DESCRIPTION
**Pull request by mw918**
Fix porttest bug where Windows paths with / give false negatives

### CODE REVIEW

File: fvtest/porttest/si.cpp [hunk 1]
```diff
@@ -164,6 +164,13 @@ TEST(PortSysinfoTest, sysinfo_test0)
 		outputErrorMessage(PORTTEST_ERROR_ARGS, "  Executable name test failed: (rc = %d).\n", rc);
 		goto done;
 	}
+	
+	#if defined(OMR_OS_WINDOWS)
+	for (char* cursor = argv0; *cursor != '\0'; ++cursor) {
+		if (*cursor == '/') {
+			*cursor = '\\';
+		}
+	}
```
Comment (fjeremic) [inline fvtest/porttest/si.cpp]: I think you're missing the `#endif` here. The ifdef should also begin a column 0, so:

```
#if defined(OMR_OS_WINDOWS)
	for (char* cursor = argv0; *cursor != '\0'; ++cursor) {
		if (*cursor == '/') {
			*cursor = '\\';
		}
	}
#endif /* defined(OMR_OS_WINDOWS) */
```

Also there seems to already be an `#if defined(OMR_OS_WINDOWS)` on line 155. Could we just move this for loop after the check there?

**Comment by fjeremic** [comment]
@genie-omr build all

**Comment by fjeremic** [comment]
Previous build:
https://ci.eclipse.org/omr/job/PullRequest-win_x86-64/1956/console
```
 13: [----------] 30 tests from PortSysinfoTest
 13: 		omrsysinfo_get_groupname returns -1.
 13: 		If this is a supported platform, consider this as a failure
 13: 		omrsysinfo_get_load_average returns -1.
 13: 		If this is a supported platform, consider this as a failure
 13: 			This platform does not support the limit iterator
 13: [----------] 30 tests from PortSysinfoTest (12812 ms total)
```

Current build launched from this PR:
https://ci.eclipse.org/omr/job/PullRequest-win_x86-64/1957/console
```
 13: [----------] 31 tests from PortSysinfoTest
 13: 		omrsysinfo_get_groupname returns -1.
 13: 		If this is a supported platform, consider this as a failure
 13: 		omrsysinfo_get_load_average returns -1.
 13: 		If this is a supported platform, consider this as a failure
 13: 			This platform does not support the limit iterator

 13: [----------] 31 tests from PortSysinfoTest (14375 ms total)
```

The new test is now executing and it is passing, at least on Jenkins. Waiting for the AppVeyor build now. @rwy0717 / @youngar could we get an approval from either of you?

File: fvtest/porttest/si.cpp [hunk 2]
```diff
@@ -157,6 +157,14 @@ TEST(PortSysinfoTest, sysinfo_test0)
 	if (0 == strncmp(argv0, "./", 2)) {
 		argv0 = &argv0[2];
 	}
+	/* When this test is invoked from a Windows machine, the input in `argv` may have forward or back slashes. To avoid false
+     * negatives we normalize the path names to use forward slashes always.
+     */
```
Comment (rwy0717) [inline fvtest/porttest/si.cpp]: Indentation is using 5 spaces, please use 1 tab and 1 space.
```suggestion
	 * negatives we normalize the path names to use forward slashes always.
	 */
```

**Comment by rwy0717** [comment]
Besides a really minor whitespace nitpick, lgtm. I guess this means the normalize path helper from the windows shell API didn't do the trick?

**Comment by youngar** [comment]
@mw918 Thank you for fixing this broken test!

**Comment by fjeremic** [comment]
> Besides a really minor whitespace nitpick, lgtm. I guess this means the normalize path helper from the windows shell API didn't do the trick?

Yeah it was failing to link. I tried locally on a unit test though and it was not changing the slashes. It only seemed to handle '.' and '..' in the paths.

**Comment by fjeremic** [comment]
@genie-omr build all

**Comment by fjeremic** [comment]
@genie-omr build zos

**Comment by fjeremic** [comment]
@genie-omr build all

**Pull request merged**